pennylane/gradients/hadamard_gradient.py

```diff
@@ -183,6 +162,32 @@ def hadamard_grad(
     .. details::
         :title: Usage Details
 
+        This gradient method can work with any operator that has a generator:
+
+        .. code-block:: pycon
+
+            >>> dev = qml.device('default.qubit')
+            >>> @qml.qnode(dev)
+            ... def circuit(x):
+            ...     qml.evolve(qml.X(0) @ qml.X(1) + qml.Z(0) @ qml.Z(1) + qml.H(0), x )
+            ...     return qml.expval(qml.Z(0))
```

Comment:
```suggestion
            ...     return qml.expval(qml.Z(0))
            ...
```
pycon env needs a blank line after def so that users can successfully copy paste into interative REPL